app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantListViewModel.kt

```diff
@@ -35,27 +40,37 @@ class PlantListViewModel @ViewModelInject internal constructor(
     @Assisted private val savedStateHandle: SavedStateHandle
 ) : ViewModel() {
 
-    val plants: LiveData<List<Plant>> = getSavedGrowZoneNumber().switchMap {
-        if (it == NO_GROW_ZONE) {
+    private val growZone: MutableStateFlow<Int> = MutableStateFlow(
+        savedStateHandle.get(GROW_ZONE_SAVED_STATE_KEY) ?: NO_GROW_ZONE
+    )
+
+    val plants: LiveData<List<Plant>> = growZone.flatMapLatest { zone ->
+        if (zone == NO_GROW_ZONE) {
             plantRepository.getPlants()
         } else {
-            plantRepository.getPlantsWithGrowZoneNumber(it)
+            plantRepository.getPlantsWithGrowZoneNumber(zone)
+        }
+    }.asLiveData()
+
+    init {
+
+        // When `growZone` changes, store the new value in savedStateHandle
+        viewModelScope.launch {
```

Comment:
As you're calling `collect`, you can consume the elements in the `collect`'s lambda block instead of using the `onEach` operator:

```
viewModelScope.launch {
    growZone.collect { newGrowZone ->
        savedStateHandle.set(GROW_ZONE_SAVED_STATE_KEY, newGrowZone)
    }
}
```

Alternatively, you can avoid creating a new coroutine using the `launchIn` terminal operator. In that case, `onEach` is needed because `launchIn` doesn't take a lambda to consume the new element in the flow, it takes a `CoroutineScope` that uses to create a coroutine internally. 

```
growZone.onEach { newGrowZone ->
    savedStateHandle.set(GROW_ZONE_SAVED_STATE_KEY, newGrowZone)
}.launchIn(viewModelScope)
```

Both options are equivalent :) use the one you like most

Reply:
Chose the former approach and also added a description comparing the two based on the discussion here